### PR TITLE
maintainers: create sage team

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Maintaining all the packages whose updates might break some aspect of
sage.

For reference:
https://github.com/NixOS/nixpkgs/pull/103810#issuecomment-727536510

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by timokau** [comment]
@7c6f434c do you want to be a member of that team?

**Comment by timokau** [comment]
For some reason my local eval check is failing, but I don't see what is wrong. Other teams are used in the same way. Checking if ofBorg also gives an error.

**Comment by 7c6f434c** [comment]
>@7c6f434c do you want to be a member of that team?

Right now I will not be a good one… I have forgotten all the Sage stuff from my failures, and I am usually a bit behind on reacting to things…

But could join with a hope of providing a fallback when fixing some of the dependencies.

**Review by jonringer — APPROVED** [review]

File: pkgs/development/libraries/glpk/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/glpk/default.nix:76]:
```
  meta = with stdenv.lib; {
    description = "The GNU Linear Programming Kit";
    longDescription =
      '' The GNU Linear Programming Kit is intended for solving large
         scale linear programming problems by means of the revised
         simplex method.  It is a set of routines written in the ANSI C
         programming language and organized in the form of a library.
      '';
    homepage = "https://www.gnu.org/software/glpk/";
    license = stdenv.lib.licenses.gpl3Plus;
    maintainers = with maintainers; [ bjg ] ++ teams.sage.members;
    platforms = platforms.all;
  };
```

This fixes the current eval error. Also the change in line two where you added lib can be reverted with this.

File: pkgs/development/libraries/linbox/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/linbox/default.nix:77]:
```suggestion
    maintainers = with stdenv.lib.maintainers; [ ] ++ teams.sage.members;
```
or just ``stdenv.lib.teams.sage.members``

File: pkgs/development/python-modules/sympy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sympy/default.nix:34]:
```suggestion
    maintainers = with maintainers; [ lovek323 ] ++ teams.sage.members;
```

File: pkgs/development/python-modules/sympy/1_5.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sympy/1_5.nix:34]:
```suggestion
    maintainers = with maintainers; [ lovek323 ] ++ teams.sage.members;
```

File: pkgs/development/libraries/glpk/default.nix
Comment (timokau) [reply pkgs/development/libraries/glpk/default.nix:76]: Oh, I had assumed `teams` was an attribute of `maintainers`, not of `lib` directly. Thanks!

File: pkgs/development/libraries/linbox/default.nix
Comment (timokau) [reply pkgs/development/libraries/linbox/default.nix:77]: As mentioned in the other thread, I was confused myself and thought it was `lib.maintainers.teams`. Then the `with maintainers` would have made it easier to add further maintainers. Changed now.

File: pkgs/development/python-modules/sympy/default.nix
Comment (timokau) [reply pkgs/development/python-modules/sympy/default.nix:34]: Thanks!

**Comment by timokau** [comment]
> But could join with a hope of providing a fallback when fixing some of the dependencies.

More eyes are always better. I'll probably also take on more of an advisory role, at least for the near future. The biggest risk I see with teams is dilution of responsibility, but it should be manageable.

Does https://github.com/NixOS/nixpkgs/pull/104083/commits/b7319d2d35e5f50e777723b3b1c448ad05dc9e9a look good do you @7c6f434c? 

@omasanori please also explicitly approve the PR if you are fine with it. I don't want to add anybody against their will ;)

**Review by omasanori — APPROVED** [review]
@timokau Sorry for my late response.

I am glad to join the team to help the sustainability of nixpkgs mathematical and scientific computing stack.

**Comment by 7c6f434c** [comment]
@timokau I am definitely OK with not being maintainer on those things. I am also OK as being listed in the team; I guess for now it has the benefit of me automatically being notified of changes approved by @omasanori so I can merge them (I read what I merge, but I always trust other people's claim of testing, and mostly trust some people's claims of impact evaluation). If both of us have spotty availability I guess it's better to flip the coin twice?

**Comment by timokau** [comment]
> Sorry for my late response.

I'd say that was quite fast :)

> I read what I merge, but I always trust other people's claim of testing, and mostly trust some people's claims of impact evaluation.

Similar for me.

> I am also OK as being listed in the team

Alright, let's go ahead with this then!

**Comment by 7c6f434c** [comment]
I think you ended up not adding me, at least judging from GitHub diff; I am fine with it if it was your intent; otherwise I guess I wonder if it is mixing up what you committed and what you did not, or GitHub failing at diffs, because the latter case would be… something to know, but no longer unimaginable

**Comment by timokau** [comment]
Right, I didn't stage my changes to `team-list.nix` when making the second commit! Thanks for noticing, I'll fix that up.

**Pull request merged**
